types/json_marshaler.go

```diff
@@ -0,0 +1,84 @@
+package types
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/proto"
+)
+
+var _ codec.JSONMarshaler = (*JsonMarshaller)(nil)
+
+// JsonMarshaller that omits empty values
+type JsonMarshaller struct {
```

Comment:
A specific struct and file name would highlight its concrete implementation as a protobuf JSON marshaller and avoid assuming a default amino Json marshaller, i.e. `ProtoJsonMarshaller`.